src/test/time_utils.cpp

```diff
@@ -0,0 +1,74 @@
+// Copyrighted (C) 2015-2016 Antinet.org team, see file LICENCE-by-Antinet.txt
+
+#include "../libs0.hpp"
+#include "gtest/gtest.h"
+#include "../utils/time_utils.hpp"
+
+using namespace time_utils;
+
+// TODO write unit test that meet "Leap second" exception
+
+std::string gen_formated_date(int year, int month, int day) {
+	std::time_t t = std::time(NULL);
+	std::tm tm = *std::localtime(&t);
+
+	// -1900 because year start from 1900
+	tm.tm_year = year - 1900;
+	// -1 because [0-11]
+	tm.tm_mon = month-1;
+	tm.tm_mday = day;
+
+	// Daylight Saving Time flag. The value is positive if DST is in effect
+	tm.tm_isdst=1;
+	//tm.tm_gmtoff = 3600;
+
+	std::time_t test_time = mktime( &tm );
+	return timepoint_to_readable(test_time);
+}
+
+TEST(time_utils, check_readable_time_length) {
+
+	std::string formated = gen_formated_date(1023,6,1);
+	EXPECT_TRUE(formated.length() == 25);
+
+	formated = gen_formated_date(1923,12,30);
+	EXPECT_TRUE(formated.length() == 25);
+
+	formated = gen_formated_date(2923,10,20);
+	EXPECT_TRUE(formated.length() == 25);
+}
+
+TEST(time_utils, check_readable_time_regex) {
+
+	//example match: 2017-01-02T09:10:53+00:00
+	const std::string year_regexp(R"((?:[1-9]\d{3})");
+	const std::string month_regexp(R"((?:(?:0[1-9]|1[0-2]))");
+	const std::string day_regexp(R"((?:0[1-9]|1\d|2[0-8])|(?:0[13-9]|1[0-2])-(?:29|30)|(?:0[13578]|1[02])-31)|(?:[1-9]\d(?:0[48]|[2468][048]|[13579][26])|(?:[2468][048]|[13579][26])00)-02-29))");
+	const std::string hour(R"((?:[01]\d|2[0-3]):[0-5]\d:[0-5]\d)");
+	const std::string timeshift(R"((?:Z|[+-][01]\d:[0-5]\d))");
+	std::regex re("^" + year_regexp + "-" + month_regexp + "-" + day_regexp + "T" + hour + timeshift + "$");
```

Comment:
Added functionality to choose time zone.
Example of use:
int year = 2017;
int month = 12;
int day = 1;
int hour = 15;
int min = 10;
int sec= 10;
string zone =  "Asia/Pyongyang"
time_t time = gen_exact_date(year,month,day,hour,min,sec);
std::string formated_string (time, zone);
// or
//auto formated_str = time_t_to_readable(gen_exact_date(2017,10,1,0,0,0), "Asia/Pyongyang");